src/mlpack/tests/load_save_test.cpp

```diff
@@ -1727,6 +1731,78 @@ BOOST_AUTO_TEST_CASE(CategoricalNontransposedCSVLoadTest04)
     remove("test.csv");
 }
 
+/**
+ * Test For MinMax Scaler Class.
+ */
+BOOST_AUTO_TEST_CASE(MinMaxScalerTest)
+{
+    arma::mat dataset = "-1 -0.5 0 1;"
+                        "2 6 10 18;";
+    arma::mat scaleddataset;
+    arma::mat temp;
+    arma::mat scaled = "0 0.2500 0.5000 1.000;"
+                       "0 0.2500 0.5000 1.000;";
+    data::MinMaxScaler scale;
+    scale.Transform(dataset, scaleddataset);
+    scale.InverseTransform(scaleddataset, temp);
+    CheckMatrices(scaleddataset, scaled);
+    CheckMatrices(dataset, temp);
+}
+
+/**
+ * Test For MaxAbs Scaler Class.
+ */
+BOOST_AUTO_TEST_CASE(MaxAbsScalerTest)
+{
+    arma::mat dataset = "-1 -0.5 0 1;"
+                        "2 6 10 18;";
+    arma::mat scaleddataset;
+    arma::mat temp;
+    arma::mat scaled = "-1 -0.5 0 1;"
+                       "0.1111111111 0.3333333333 0.55555556 1.0000;";
+    data::MaxAbsScaler scale;
+    scale.Transform(dataset, scaleddataset);
+    scale.InverseTransform(scaleddataset, temp);
+    CheckMatrices(scaleddataset, scaled);
+    CheckMatrices(dataset, temp);
+}
+
+/**
+ * Test For Standard Scaler Class.
+ */
+BOOST_AUTO_TEST_CASE(StandardScalerTest)
+{
+    arma::mat dataset = "-1 -0.5 0 1;"
+                        "2 6 10 18;";
+    arma::mat scaleddataset;
+    arma::mat temp;
+    arma::mat scaled = "-1.18321596 -0.50709255  0.16903085 1.52127766;"
+                       "-1.18321596 -0.50709255  0.16903085 1.52127766;";
+    data::StandardScaler scale;
+    scale.Transform(dataset, scaleddataset);
+    scale.InverseTransform(scaleddataset, temp);
+    CheckMatrices(scaleddataset, scaled);
+    CheckMatrices(dataset, temp);
+}
+
+/**
+ * Test For MeanNormalization Scaler Class.
+ */
+BOOST_AUTO_TEST_CASE(MeanNormalizationTest)
+{
+    arma::mat dataset = "-1 -0.5 0 1;"
+                        "2 6 10 18;";
+    arma::mat scaleddataset;
+    arma::mat temp;
+    arma::mat scaled = "-0.43750000000 -0.187500000 0.062500000 0.562500000;"
+                       "-0.43750000000 -0.187500000 0.062500000 0.562500000;";
+    data::MeanNormalization scale;
+    scale.Transform(dataset, scaleddataset);
+    scale.InverseTransform(scaleddataset, temp);
+    CheckMatrices(scaleddataset, scaled);
+    CheckMatrices(dataset, temp);
+}
```

Comment:
You should probably also add some tests where you try and "break" your scalers, by passing in empty data, or passing in data where the scale factor will be 0 for some dimensions, or something like this. :)

Reply:
Ooh, another fun one might be passing in the same matrix as input and output and making sure it still works correctly.

Reply:
> You should probably also add some tests where you try and "break" your scalers, by passing in empty data, or passing in data where the scale factor will be 0 for some dimensions, or something like this. :)

Added a test which has empty data, and also which has dimension with scale factor 0

Reply:
> Ooh, another fun one might be passing in the same matrix as input and output and making sure it still works correctly.

That was a nice suggestion, added accordingly and glad to see that it passes without any issues